### PR TITLE
docs(core): additional documentation and examples

### PR DESCRIPTION
**Pull request by sinui0**
This PR adds more documentation to the core crate, including some examples. To simplify the example for verifying a presentation I added a method for fetching the key.

### CODE REVIEW

File: crates/core/src/attestation/proof.rs [hunk 1]
```diff
@@ -98,6 +103,10 @@ impl BodyProof {
         Ok(BodyProof { body, proof })
     }
```
Comment (yuroitaki) [inline crates/core/src/attestation/proof.rs:105]: Need documentation comment for this func?

**Review by yuroitaki — COMMENTED** [review]
Nice! just one small comment.

File: crates/core/src/presentation.rs [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
+//! > that is synthesized from, but do not contain, the original verifiable
+//! > credentials (for example, zero-knowledge proofs).
+//!
+//! Instead of a credential, a presentation in this context is a proof of an
+//! attestation from a Notary along with additional selectively disclosed
+//! information about the TLS connection such as the server's identity and the
+//! application data communicated with the server.
+//!
+//! A presentation is self-contained and can be verified by a Verifier without
+//! needing access to external data. The Verifier need only check that the key
```
Comment (th4s) [inline crates/core/src/presentation.rs:20]: Is this colloquial? Shouldn't this be
```
The Verifier only needs to check ...
```

**Review by th4s — APPROVED** [review]
Nice doc :rocket:

File: crates/core/src/attestation/proof.rs (hunk 1 above)
Comment (yuroitaki) [reply crates/core/src/attestation/proof.rs:105]: oh weird, vscode plugin acting weird, i'm referring to L126

**Pull request merged**